File: Python/01 - 10/07 - eulerSeven.py
def is_prime(n):
    from math import sqrt
    if n % 2 == 0 and n > 2:
        return False
    return all(n % i for i in range(3, int(sqrt(n)) + 1, 2))

def main():
    counter, number = 6, 13
    while counter < 10001:
        number += 1
        test_prime = is_prime(number)
        if test_prime:
            counter += 1
    print(number, "is the 10001st prime number")

if __name__ == "__main__":
    main()

File: C++/01 - 10/07 - eulerSeven.cpp
#include <iostream>

using namespace std;

int main() {
	int isPrime(int number);
	bool running = true;
	int counter = 0, num = 2;
	while (running == true) {
		if (isPrime(num) == 1) {
			counter++;
		}
		if (counter == 10001) {
			running = false;
			cout << num << " is the 10001 prime number.\n";
		}
		num++;
	}
}

// Optimized prime checker from my C programs.
int isPrime(int number) {
    if (number > 2 && number % 2 == 0) return 0;
    for (int i = 3; i < sqrt(number) + 1; i++) {
        if (i % 2 == 0)
        continue;
        if (number % i == 0) return 0;
    }
    return 1;
}